aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group/Program.cs

```diff
@@ -0,0 +1,173 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.OpenApi;
+using Repository;
+using Data;
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+// Add TodoRepository to the container.
+builder.Services.AddScoped<TodosRepo>();
+// Add NoteRepository to the container.
+builder.Services.AddScoped<NotesRepo>();
+// Add InMemoryDatabase to the container.
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+{
+    options.UseInMemoryDatabase("InMemoryDb");
+});
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    // localhost:{port}/swagger
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+
+app.MapGet("/", () => "Hello World!");
+
+// group apis
+app.MapGroup("/todos").MapTodosApi();
```

Comment:
Do we want to add any endpoint filters or endpoint metedata to these groups using methods like. `AddEndpointFilter`, `WithTags` and/or `WithOpenApi()`? All of these should work just like they would when applied directly to the endpoint, but will be run with a lower precedence for the entire group.